plugin/action/mask/mask.go

```diff
@@ -400,9 +434,13 @@ func (p *Plugin) Do(event *pipeline.Event) pipeline.ActionResult {
 	locApplied := false
 
 	p.valueNodes = p.valueNodes[:0]
-	p.valueNodes = getValueNodeList(root, p.valueNodes)
+	p.valueNodes = getValueNodeList(root, p.valueNodes, p.config.IgnoredFields_)
 	for _, v := range p.valueNodes {
 		value := v.AsBytes()
+		if p.config.SkipMismatched && !p.matchRe.Match(value) {
```

Comment:
Some masks can have only match rules and no regex but here all masks will be skipped if `SkipMismatched` is true, no matter if they have regex or not. Maybe it would be better to skip only masks with regex in that case.

Reply:
I added a skip for empty masks and wrote TestWithEmptyRegex